pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -163,23 +163,43 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 			reqLogger.Error(err1, "Could not update error conditions")
 		}
 		return *recResult, err
-	} else {
-		if infinispan.RemoveCondition("preliminaryChecksFailed") {
-			err1 := r.client.Status().Update(context.TODO(), infinispan)
-			reqLogger.Error(err1, "Could not update error conditions")
-		}
+	}
+	if infinispan.RemoveCondition("preliminaryChecksFailed") {
+		err1 := r.client.Status().Update(context.TODO(), infinispan)
+		reqLogger.Error(err1, "Could not update error conditions")
```

Comment:
@rigazilla maybe we can introduce upodateInfinispan func with reconcile on error and message parameters as we have about 10 infinispan's updates in the code?

Reply:
I'm not sure, I can see the pros of saving some lines of code, but I feel the code is more readable with the explicit api calls.

Reply:
Yes, true, but all coding tech like inheritance, funcs, templates, etc. make code less readable but give us more advanced things and reduce the chance of errors. But we can keep this "as is" as reduced number of code lines are not so large.

P.S. I saw at least two Objects update without error handling. Is it by design or something missing?

Reply:
yeah I also noticed that, I'll do a quick check

Reply:
I've added some error handling